addons/image/webp.c

```diff
@@ -0,0 +1,174 @@
+/* Allegro wrapper routines for libwebp
+ * by Sebastian Krzyszkowiak (dos@dosowisko.net).
+ */
+
+
+#include <webp/decode.h>
+#include <webp/encode.h>
+
+#include "allegro5/allegro.h"
+#include "allegro5/allegro_image.h"
+#include "allegro5/internal/aintern_image.h"
+
+#include "iio.h"
+
+ALLEGRO_DEBUG_CHANNEL("image")
+
+
+/*****************************************************************************
+ * Loading routines
+ ****************************************************************************/
+
+
+static ALLEGRO_BITMAP *load_from_buffer(const uint8_t* data, size_t data_size,
+   int flags)
+{
+   ALLEGRO_BITMAP *bmp;
+   ALLEGRO_LOCKED_REGION *lock;
+   bool premul = !(flags & ALLEGRO_NO_PREMULTIPLIED_ALPHA);
+
+   WebPDecoderConfig config;
+   WebPInitDecoderConfig(&config);
+
+   if (WebPGetFeatures(data, data_size, &config.input) != VP8_STATUS_OK) {
+      ALLEGRO_ERROR("Could not read WebP stream info\n");
+      return NULL;
+   }
+
+   bmp = al_create_bitmap(config.input.width, config.input.height);
+   if (!bmp) {
+      ALLEGRO_ERROR("al_create_bitmap failed while loading WebP.\n");
+      return NULL;
+   }
+
+   lock = al_lock_bitmap(bmp, ALLEGRO_PIXEL_FORMAT_ABGR_8888_LE,
+      ALLEGRO_LOCK_WRITEONLY);
+
+   config.output.colorspace = premul ? MODE_rgbA : MODE_RGBA;
+   config.output.u.RGBA.rgba = (uint8_t*)lock->data;
+   config.output.u.RGBA.stride = lock->pitch;
+   config.output.u.RGBA.size = lock->pitch * config.input.height;
+   config.output.is_external_memory = 1;
+
+   if (WebPDecode(data, data_size, &config) != VP8_STATUS_OK) {
```

Comment:
Now, I am convinced that libwebp is very easy to use compared to libpng...😏

Reply:
And that's "advanced API" - it could be even easier if not for premultiplied alpha :D 
https://developers.google.com/speed/webp/docs/api#simple_decoding_api